projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -234,11 +260,17 @@ public static function get_connections( $run_tests = false ) {
 		}
 
 		$path = add_query_arg(
-			array( 'test_connections' => $run_tests ),
+			$args,
 			sprintf( '/sites/%d/publicize/connections', $site_id )
 		);
 
-		$response = Client::wpcom_json_api_request_as_user( $path, 'v2', array( 'method' => 'GET' ) );
+		$include = $args['include'] ?? '';
+
+		$blog_or_user = 'site-wide' === $include ? 'blog' : 'user';
+
+		$callback = array( Client::class, "wpcom_json_api_request_as_{$blog_or_user}" );
```

Comment:
Do we want to always request as blog, and then filter by user? That means the caching can be based on the site's connections.

Reply:
Yeah, I realise we would need to filter the connections we receive (or have stored in the cache) but it would mean that the caching is simpler I think?

Reply:
We want to consume the endpoint directly in Jetpack sites and thus keeping the endpoint free from any caching logic is better and then we can apply caching on top of that for Jetpack sites. That caching class and script data will always request all the connections by default and then pass only the appropriate connections via the initial state. Then there are connection test results which we don't cache anywhere and it uses the same endpoint and should always return the connections for the user and the shared ones.